topic_resources/rspec.md

```diff
@@ -0,0 +1,58 @@
+### RSPEC
+[Rspec](rspec.info)
+
+[Additional Tutorial Info](http://code.tutsplus.com/tutorials/ruby-for-newbies-testing-with-rspec--net-21297)
+
+#### Setting Up
+Install Rspec
+``` gem install rspec ```
+
+For each class file that we create, we create one spec file.
+
+Create the ```chair_spec.rb``` test file.
+```
+require 'spec_helper'
+
+describe Chair do
+
+end
+```
+
+Use a ```before``` block to run a specific block of code before every test is run. In this before block example, we will ensure we have an instance of our chair object before we execute these tests.
+
+```
+before :each do
+    @chair = Chair.new "Spinny", "Yellow", :height
+end
+```
+
+Use the describe block to specify actions of a method.
+```
+describe "#new" do
+    it "takes three parameters and returns a Chair object" do
+        @chair.should be_an_instance_of Chair
+    end
+end
+```
+This test simply confirms that we have indeed made a Chair object.
+
+You’ll notice I’ve used the string ```#new``` this is a convention in Ruby to refer to instance methods like this: ```ClassName#methodName``` Since we have the class name in our top-level describe, we’re just putting the method name here.
+
+Also notice the grammar we use within the block here: ```object.should do_something```. Ninety-nine percent of your tests will take this form: you have an object, and you start by calling ```should``` or ```should_not``` on the object.
```

Comment:
Another note: the most recent version of rspec (3.2) tends to use the `expect(object).to eq` syntax rather than the .`should` syntax. (The blog post referenced is from 2011, so those changes may not have been implemented then) Not sure if this is an important distinction in the context of a first-time intro to rspec, but presumably the students will be using the newest gems.